components/Components/SingleAddress.tsx

```diff
@@ -63,81 +208,140 @@ const SingleAddress: React.FunctionComponent<SingleAddressProps> = ({
     });
   };
 
+  function onCopy() {
+    addLastSnackbar({
+      message: ufvk
+        ? (translate('seed.tapcopy-ufvk-message') as string)
+        : (translate('history.addresscopied') as string),
+      duration: SnackbarDurationEnum.short,
+    });
+  }
+
   const doNothing = () => {
     addLastSnackbar({
       message: 'Unimplemented option',
       duration: SnackbarDurationEnum.short,
     });
   };
 
-  const doShare = async () => {
-    if (qrCodeRef.current && qrCodeRef.current.capture) {
-      let changed: boolean = false;
-      if (security.foregroundApp) {
-        // deactivate temporarily this
-        changed = true;
-        const newSecurity = {
-          startApp: security.startApp,
-          foregroundApp: false,
-          sendConfirm: security.sendConfirm,
-          seedUfvkScreen: security.seedUfvkScreen,
-          rescanScreen: security.rescanScreen,
-          settingsScreen: security.settingsScreen,
-          changeWalletScreen: security.changeWalletScreen,
-          restoreWalletBackupScreen: security.restoreWalletBackupScreen,
-        } as SecurityType;
-        setSecurityOption(newSecurity);
-      }
-      try {
-        const uri = await qrCodeRef.current.capture(); // Capture the QR code as an image URI
-        const shareOptions = {
-          title: 'QR',
-          url: uri,
-          type: 'image/png',
-        };
-        await Share.open(shareOptions);
-      } catch (error) {
-        // https://github.com/react-native-share/react-native-share/issues/1664
-        console.log('Error sharing QR code:', error);
-      }
-      if (changed) {
-        // activate again in 5 seconds
-        setTimeout(() => {
-          const newSecurity = {
-            startApp: security.startApp,
-            foregroundApp: true,
-            sendConfirm: security.sendConfirm,
-            seedUfvkScreen: security.seedUfvkScreen,
-            rescanScreen: security.rescanScreen,
-            settingsScreen: security.settingsScreen,
-            changeWalletScreen: security.changeWalletScreen,
-            restoreWalletBackupScreen: security.restoreWalletBackupScreen,
-          } as SecurityType;
-          setSecurityOption(newSecurity);
-        }, 5 * 1000);
-      }
-    }
-  };
-
   return (
     <View style={{ flexDirection: 'column', width: '100%' }}>
       <ScrollView
         style={{ width: '100%' }}
         contentContainerStyle={{
           alignItems: 'center',
         }}>
-        {(ufvk || (address && address.address !== (translate('receive.noaddress') as string))) ? (
+        {ufvk || (address && address.address !== (translate('receive.noaddress') as string)) ? (
           <>
-            <View style={{ flexDirection: 'row', justifyContent: 'center', alignItems: 'center', marginTop: 10, marginBottom: 5 }}>
-              {mode === ModeEnum.advanced && address && address.addressKind === AddressKindEnum.u && (
+            {!isBasic && address && !isUnified && (
```

Comment:
I think, those conditions are the opposite:
- address.addressKind === AddressKindEnum.u
- !isUnified